pandas/_typing.py

```diff
@@ -65,9 +65,19 @@
 
     from pandas.io.formats.format import EngFormatter
 
+    ScalarLike_co = Union[
+        int,
+        float,
+        complex,
```

Comment:
I think we could shorten `int | float | complex` to just `complex` (see [flake8-pyi](https://github.com/PyCQA/flake8-pyi)'s Y041)

Reply:
I think the union is easier to understand (at least for me), but no strong opinion